src/metabase/logger.clj

```diff
@@ -47,14 +52,89 @@
 
 (defonce ^:private has-added-appender? (atom false))
 
+(defn context
+  "Get global logging context."
+  ^LoggerContext []
+  (LogManager/getContext (classloader/the-classloader) false))
+
+(defn configuration
+  "Get global logging configuration"
+  ^Configuration []
+  (.getConfiguration (context)))
+
 (when-not *compile-files*
   (when-not @has-added-appender?
     (reset! has-added-appender? true)
-    (let [^LoggerContext ctx (LogManager/getContext (classloader/the-classloader) false)
-          config             (.getConfiguration ctx)
-          appender           (metabase-appender)]
+    (let [appender (metabase-appender)
+          config      (configuration)]
       (.start appender)
       (.addAppender config appender)
       (doseq [[_ ^LoggerConfig logger-config] (.getLoggers config)]
         (.addAppender logger-config appender (.getLevel logger-config) (.getFilter logger-config))
-        (.updateLoggers ctx)))))
+        (.updateLoggers (context))))))
+
+;;; Custom loggers
+
+(defn logger-name
+  "Get string name from symbol or ns"
+  ^String [a-namespace]
+  (if (instance? clojure.lang.Namespace a-namespace)
+    (name (ns-name a-namespace))
+    (name a-namespace)))
```

Comment:
Maybe this actually like 3 PRs because this seems unrelated to either the `defendpoint2` stuff or the new SerDes API endpoints. I'm a big +1 on test tooling improvements but I would split significant changes out into their own PR so they can be reviewed separately. It's always good to explain the reasoning for the changes too